### ISSUE TITLE
💅 `useReactCompiler` reports `CapitalizedCalls`, which upstream `eslint-plugin-react-hooks` ships off by default, with no way to opt out

### ISSUE THREAD

**Issue by viktor-horacek** [biomejs/biome#11319]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.8
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  JS_RUNTIME_VERSION:           v24.19.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.3.14

Biome Configuration:
  Status:                       Loaded successfully.
  Path:                         biome.json
  Formatter enabled:            false
  Linter enabled:               true

Linter:
  Enabled rules:
    nursery/useReactCompiler
```

</details>

### Rule name

lint/nursery/useReactCompiler

### Playground link

Not reproducible in the playground: the rule only activates when the nearest package.json declares React 19+, and the playground has no package.json. A fully self-contained three-file reproduction is included under Expected result.

### Expected result

No diagnostic for `TEST_ID.MENU_OPTION("light")`. It returns a string used as a `data-testid`; it is not a component.

More precisely: enabling `useReactCompiler` should not surface the `CapitalizedCalls` validation by default, because the upstream rule it corresponds to (`react-hooks/capitalized-calls`) is **off in every upstream preset**.

## Reproduction

`package.json` (required — this is what gates the rule on):

```json
{
  "name": "biome-usereactcompiler-repro",
  "version": "1.0.0",
  "private": true,
  "type": "module",
  "dependencies": {
    "react": "^19.0.0",
    "react-dom": "^19.0.0"
  }
}
```

`biome.json`:

```json
{
  "$schema": "https://biomejs.dev/schemas/2.5.8/schema.json",
  "formatter": { "enabled": false },
  "linter": {
    "enabled": true,
    "rules": {
      "preset": "none",
      "nursery": {
        "useReactCompiler": "error"
      }
    }
  }
}
```

`App.tsx` — line 12 is the reported case, line 13 is the control (identical code, camelCase name):

```tsx
const TEST_ID = {
  MENU_OPTION: (name: string) => `menu-option-${name}`,
};

const testId = {
  menuOption: (name: string) => `menu-option-${name}`,
};

export function ThemeMenu() {
  return (
    <ul>
      <li data-testid={TEST_ID.MENU_OPTION("light")}>Light</li>
      <li data-testid={testId.menuOption("dark")}>Dark</li>
    </ul>
  );
}
```

```bash
bun install
bunx @biomejs/biome@2.5.8 check .
```

## Actual result

```
App.tsx:12:21 lint/nursery/useReactCompiler ━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  × This capitalized function is called directly.

    10 │ 	return (
    11 │ 		<ul>
  > 12 │ 			<li data-testid={TEST_ID.MENU_OPTION("light")}>Light</li>
       │ 			                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    13 │ 			<li data-testid={testId.menuOption("dark")}>Dark</li>
    14 │ 		</ul>

  i React treats capitalized functions as components, and components should be rendered with JSX so their hooks and lifecycle are handled correctly.

Checked 3 files in 10ms. No fixes applied.
Found 1 error.
```

The camelCase control on line 13 is not reported, confirming the name casing is the only trigger.

## Analysis

This is a **preset-parity and configurability** problem rather than a detection problem:

1. **Detection itself matches upstream.** With `eslint-plugin-react-hooks@7.1.1` and `react-hooks/capitalized-calls` explicitly turned on, the same line is reported. Biome's port is faithful — the problem is what is enabled and what can be turned off.
2. **Upstream ships this check off by default.** `capitalized-calls` is a standalone rule in the plugin, absent from both `recommended` and `recommended-latest` (enumerated from `eslint-plugin-react-hooks@7.1.1`). Running `recommended-latest` over the exact `App.tsx` above reports nothing (exit 0).
3. **Upstream also has an allowlist escape hatch.** The compiler environment option bundled in the plugin is `validateNoCapitalizedCalls: z.nullable(z.array(z.string()))` — off by default, an explicit list of permitted capitalized callees when on.
4. **Biome exposes neither.** `useReactCompiler`'s only option is `compilationMode`; there is no way to disable just this category or allowlist names, so the only remedy is a per-call-site `biome-ignore`.

**Why this matters in practice:** SCREAMING_CASE constant objects holding small builder functions are a widespread convention — test-id builders, route builders, i18n key builders. In a codebase using one, every call site becomes an error. For users migrating from `eslint-plugin-react-hooks` presets, enabling the Biome rule introduces errors upstream never emitted.

**Possible resolutions** (any one would work, in rough order of preference):

1. Exclude the `CapitalizedCalls` category from the reportable set by default, matching upstream presets.
2. Add an option mirroring `validateNoCapitalizedCalls` (allowlist of permitted capitalized callees).
3. Split the rule so this category can be toggled independently — already listed as "Consider splitting it into multiple rules" in #10974.

Related: #10974 (`useReactCompiler` remaining work), which also tracks "Address prefiltering heuristics problems".

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Use the playground https://biomejs.dev/playground/?lintRules=useReactCompiler&tab=settings&pane=Diagnostics

**Comment by biome-cookie** [comment]
Hello @viktor-horacek, please provide a minimal reproduction. You can use one of the following options:

- Provide a link to [our playground](https://biomejs.dev/playground), if it's applicable.
- Provide a link to GitHub repository. To easily create a reproduction, you can use our interactive CLI via `npm create @biomejs/biome-reproduction`

Issues marked with `S-Needs repro` will be **closed** if they have **no activity within 3 days**.

**Comment by biome-cookie** [comment]
This issue has been automatically closed because it has been inactive for more than 3 days and it does not have a minimal reproduction.
